src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,104 @@
-""" This is a dummy example """
+import requests
+import json
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+from datetime import timedelta
+from ratelimit import limits, sleep_and_retry
+
+
+COORDINATES = {
+    "Madrid": {"latitude": 40.416775, "longitude": -3.703790},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+}
+
+VARIABLES = "temperature_2m_mean,precipitation_sum,soil_moisture_0_to_10cm_mean"
+
+
+def get_data_meteo_api(lat: float, long: float, variable: str):
+    # For the API call I only parametrized the coordinates as
+    # they are the only parameters that will vary in this case
+    API_URL_0 = f"https://climate-api.open-meteo.com/v1/climate?latitude={lat}&longitude={long}&start_date=1950-01-01&end_date=2050-12-31&models=CMCC_CM2_VHR4,FGOALS_f3_H,HiRAM_SIT_HR,MRI_AGCM3_2_S,EC_Earth3P_HR,MPI_ESM1_2_XR,NICAM16_8S&daily="
+    API_URL = API_URL_0 + variable
+    return pd.DataFrame(json.loads(api_request_manager(API_URL)))
+
+
+# This function controls that the API call limit is not exceeded
+# and also raises an error if the status is not 200
+@sleep_and_retry
```

Comment:
It's great that you're using this library instead of implementing the logic from scratch. However, after reading the documentation [documentation](https://pypi.org/project/ratelimit/), I am not sure if it's the most suitable one for our use case. It seems intended to prevent applications from making more than a certain number of queries per second when continuously calling an API, when in our case what we really want is to keep retrying until ONE call succeeds. Instead I'd use this one: https://pypi.org/project/backoff/, which by the way has quite more github stars